test/unit/validate.test.js

```diff
@@ -39,12 +39,18 @@ describe('Recurly.validate', function () {
     });
 
     it('should parse discover', function () {
-      assert(recurly.validate.cardType('3050990900242198') === 'discover');
-      assert(recurly.validate.cardType('36003010814263') === 'discover');
-      assert(recurly.validate.cardType('6011111111111117') === 'discover');
-      assert(recurly.validate.cardType('6011000990139424') === 'discover');
-      assert(recurly.validate.cardType('6229252986142899') === 'discover');
-      assert(recurly.validate.cardType('8131917256371926354') === 'discover');
+      assert(recurly.validate.cardType('6011000090139424') === 'discover');
+      assert(recurly.validate.cardType('64400000901394243') === 'discover');
+      assert(recurly.validate.cardType('650611009013942423') === 'discover');
+      assert(recurly.validate.cardType('601179999013942423') === 'discover');
+    });
+
+    it('should parse union_pay', function () {
+      assert(recurly.validate.cardType('6221261111113245') === 'union_pay');
+      assert(recurly.validate.cardType('6282000123842342') === 'union_pay');
+      assert(recurly.validate.cardType('6250941006528599') === 'union_pay');
+      assert(recurly.validate.cardType('8171999927660000') === 'union_pay');
+      assert(recurly.validate.cardType('8171999900000000021') === 'union_pay');
```

Comment:
Seems like these card numbers aren't properly testing the ranges. Let's add some that are out of range to test that they are not recognized as discover or union pay

Reply:
Sure thing. Those are test cards from the discover / union pay docs. I can add the out of range negative case checks.